lib/charms/operator_libs_linux/v0/juju_systemd_notices.py

```diff
@@ -256,6 +292,16 @@ def _dbus_path_to_name(path: str) -> str:
     return name
 
 
+@functools.lru_cache(maxsize=32)
```

Comment:
Why 32? Would 1 be sufficient, given there are no arguments? (We could in future use `functools.cache`, but that's Python 3.9+ only, and this repo supports 3.8.)

Reply:
I did think about using `functools.cache`, but yeah, it wasn't added until Python 3.9+. I mostly used `32` because it felt like a good number; I haven't used `functools.lru_cache` much but I did know that I didn't want to keep reopening the `watch.yaml` file every loop :sweat_smile:

Reply:
No longer using `functools.lru_cache` as of https://github.com/canonical/operator-libs-linux/pull/128/commits/c8f0767563b3e317b91c0406c0872db930a09830.

Now instead using a global map to track observed services and aliases.